### PR TITLE
emacs: add emacsclient desktop file

### PR DESCRIPTION
**Pull request by pstch**
Add an option to enable a .desktop file for the Emacs client.

- I didn't use the `isLinux` check as `emacsclient` can be used remotely
- I'm not sure if this should be added to `programs.emacs.client` or if I should create an independent `programs.emacsclient`. If needed I can update the PR to the latter.
- Actually, I'm not sure if this PR  should be accepted, or if it's better to let the users create such desktop file themselves.

### CODE REVIEW

File: modules/misc/news.nix [hunk 1]
```diff
@@ -1245,6 +1245,16 @@ in
           A new module is available: 'services.spotifyd'.
         '';
       }
+
+      {
```
Comment (rycee) [inline modules/misc/news.nix]: Generally speaking we don't add news entries just for new options so I would suggest removing it. Alternatively, add a condition to restrict it configurations where `services.emacs` is enabled…

Comment (pstch) [reply modules/misc/news.nix]: I'll remove it.

**Comment by pstch** [comment]
Thanks for the review!

> I'm wondering, though, if this new option is not better suited for the services.emacs module since the client is mainly useful for people who user the Emacs daemon. Edit: I guess programs.emacs is fair enough due to the remote access support but is it particularly common that somebody would like to set up a desktop item for remote access but not for local?

I wasn't sure either. I actually started by writing it in `services.emacs`, then switched, but now I think it makes more sense to have it `services.emacs`, as this only provides a .desktop file (emacsclient is already available from emacs itself), which are mainly used for local daemons. Unless we provide support for configuring remote servers (and I really don't think that is needed, as you I don't think many users use a remote emacs daemon these days), I don't think it makes much sense to have it  in `programs.emacs`

TL;DR I'll apply the remaining changes, and then, if it's OK with you, move everything to `services.emacs`, rebase and update this PR.

**Comment by rycee** [comment]
Sounds good, `services.emacs` it is.

**Comment by spacekitteh** [comment]
Can this be merged?

**Comment by rycee** [comment]
@spacekitteh As far as I can tell a number of suggestions still remain to be applied and also the move to the service module.

**Comment by spacekitteh** [comment]
@pstch any progress?

**Comment by mjlbach** [comment]
I'm really interested in getting this merged. I'm working on the updates to the PR here but will defer to @pstch 
https://github.com/mjlbach/home-manager/tree/emacsclient-desktop-file

**Comment by mjlbach** [comment]
@rycee This can be closed now that #1223 is merged

**Comment by rycee** [comment]
Yup! 👍
